query/src/main/java/org/infinispan/query/dsl/embedded/impl/JPQLFilterAndConverter.java

```diff
@@ -86,8 +86,20 @@ public ObjectFilter getObjectFilter() {
       return objectFilter;
    }
 
+   public String getJPAQuery() {
+      return jpaQuery;
+   }
+
+   public Matcher getMatcher() {
+      return matcher;
+   }
+
    @Override
    public ObjectFilter.FilterResult filterAndConvert(K key, V value, Metadata metadata) {
+      if (value == null) {
+         // this is a 'pre' invocation, ignore it
```

Comment:
So is this an issue for local?  Or is that okay it is fire twice once with the old and once with the new value ?